ts bug: toConstructor(): typeof Query<ResultType, DocType, THelpers, RawDocType>;

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

6.7.2

### Node.js version

16.x

### MongoDB server version

5.x

### Typescript version (if applicable)

4.8.4

### Description

somehow this fails now suddenly:
../../node_modules/mongoose/types/query.d.ts(619,34): error TS1144: '{' or ';' expected.
../../node_modules/mongoose/types/query.d.ts(619,45): error TS1005: '>' expected.
../../node_modules/mongoose/types/query.d.ts(619,77): error TS1109: Expression expected.
../../node_modules/mongoose/types/query.d.ts(622,19): error TS1109: Expression expected.
../../node_modules/mongoose/types/query.d.ts(622,41): error TS1109: Expression expected.
../../node_modules/mongoose/types/query.d.ts(622,50): error TS1109: Expression expected.
../../node_modules/mongoose/types/query.d.ts(622,73): error TS1005: '(' expected.
../../node_modules/mongoose/types/query.d.ts(622,114): error TS1109: Expression expected.
../../node_modules/mongoose/types/query.d.ts(622,138): error TS1005: '(' expected.

the change was made already some weeks ago, but the line is definitly wrong:

toConstructor(): typeof Query<ResultType, DocType, THelpers, RawDocType>;

Query is already a Class and therefore typeof should just get removed.

![image](https://user-images.githubusercontent.com/5757263/201628524-299a3116-57a7-47be-9307-5f3e9c64f08b.png)

change line 619 to 
`toConstructor(): Query<ResultType, DocType, THelpers, RawDocType>;`

fixes it

### Steps to Reproduce

just open the query.d.ts file in your favourite IDE :)

### Expected Behavior

_No response_

--- a/types/query.d.ts
+++ b/types/query.d.ts
@@ -616,7 +616,7 @@ declare module 'mongoose' {
     then: Promise<ResultType>['then'];
 
     /** Converts this query to a customized, reusable query constructor with all arguments and options retained. */
-    toConstructor(): typeof Query<ResultType, DocType, THelpers, RawDocType>;
+    toConstructor<RetType = typeof Query>(): RetType;
 
     /** Declare and/or execute this query as an update() operation. */
     update(filter?: FilterQuery<DocType>, update?: UpdateQuery<DocType> | UpdateWithAggregationPipeline, options?: QueryOptions<DocType> | null, callback?: Callback<UpdateWriteOpResult>): QueryWithHelpers<UpdateWriteOpResult, DocType, THelpers, RawDocType>;

